lib/Fmt_ast.ml

```diff
@@ -2192,7 +2192,8 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
   | Pexp_pack me ->
       let fmt_mod m =
         Params.wrap_exp c.conf c.source ~parens:true ~loc:pexp_loc
-          (str "module " $ m $ fmt_atrs)
+          (str "module " $ m)
+        $ fmt_atrs
```

Comment:
We need an other layer of parentheses for this case:
```
let _ = f ((module struct end) [@a])
```

If `wrap_if parens` adds too much parens, `wrap_if (parens && has_attr)` should be enough because the parens will only guard the attribute.